### PR TITLE
plat-stm32mp1: repelace 2 stm32mp1 platform helper function with a driver local one

### PR DESCRIPTION
**Pull request by etienne-lms**
Adds local helper function `stm32_gpio_get_bank()` in stm32_gpio driver to remove stm32mp1 platform functions `stm32_get_gpio_bank_clock()`, `stm32_get_gpio_bank_clk()` and `stm32_get_gpio_bank_base()`.

### CODE REVIEW

**Comment by etienne-lms** [comment]
@ldebieve, @GseoC: could you have a look at these changes?

File: core/drivers/stm32_gpio.c [hunk 1]
```diff
@@ -130,9 +130,10 @@ static void get_gpio_cfg(uint32_t bank_id, uint32_t pin, struct gpio_cfg *cfg)
 static void set_gpio_cfg(uint32_t bank_id, uint32_t pin, struct gpio_cfg *cfg)
 {
 	struct stm32_gpio_bank *bank = stm32_gpio_get_bank(bank_id);
-	uint32_t exceptions = cpu_spin_lock_xsave(&gpio_lock);
+	uint32_t exceptions = 0;
 
 	clk_enable(bank->clock);
```
Comment (GseoC) [inline core/drivers/stm32_gpio.c:135]: ditto for clock error code

File: core/drivers/stm32_gpio.c [hunk 2]
```diff
@@ -77,13 +77,23 @@ static unsigned int gpio_lock;
...
 {
-	vaddr_t base = stm32_get_gpio_bank_base(bank);
-	struct clk *clk = stm32_get_gpio_bank_clk(bank);
+	struct stm32_gpio_bank *bank = NULL;
+
+	STAILQ_FOREACH(bank, &bank_list, link)
+		if (bank_id == bank->bank_id)
+			break;
+
+	return bank;
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_gpio.c:88]: ok, i'll fix.

File: core/drivers/stm32_gpio.c [hunk 3]
```diff
@@ -77,13 +77,23 @@ static unsigned int gpio_lock;
...
+}
+
+/* Save to output @cfg the current GPIO (@bank_id/@pin) configuration */
+static void get_gpio_cfg(uint32_t bank_id, uint32_t pin, struct gpio_cfg *cfg)
+{
+	struct stm32_gpio_bank *bank = stm32_gpio_get_bank(bank_id);
 
-	clk_enable(clk);
+	clk_enable(bank->clock);
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_gpio.c:96]: makes sense.

File: core/drivers/stm32_gpio.c (hunk 1 above)
Comment (etienne-lms) [reply core/drivers/stm32_gpio.c:135]: ok

File: core/arch/arm/plat-stm32mp1/main.c [hunk 4]
```diff
@@ -445,28 +445,6 @@ unsigned int stm32_get_gpio_bank_offset(unsigned int bank)
 	return bank * GPIO_BANK_OFFSET;
 }
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/main.c:419]: Yes, i plan to remove them also.

**Comment by etienne-lms** [comment]
review comments addressed (unless I missed somthing)

**Comment by ldebieve** [comment]
Fine with the last update
`Acked-by: Lionel Debieve <lionel.debieve@foss.st.com>`

**Comment by etienne-lms** [comment]
Review tags applied.
Thanks

**Pull request merged**